src/deepsparse/pipeline.py

```diff
@@ -167,11 +176,31 @@ def __init__(
         self.onnx_file_path = self.setup_onnx_file_path()
         self.engine = self._initialize_engine()
 
-    def __call__(self, threadpool=None, *args, **kwargs) -> Union[BaseModel, Future]:
-        threadpool = threadpool or self._threadpool
-        if threadpool:
-            return threadpool.submit(self._run, *args, **kwargs)
-        return self._run(*args, **kwargs)
+    def __call__(self, *args, **kwargs) -> Union[BaseModel, Future]:
+        executor = kwargs.get("executor")
+        if executor is not None:
+            kwargs.pop("executor")
+        else:
+            executor = self.executor
+
+        if self.in_dynamic_batch_mode():
+            # Blocking call in Dynamic Batch Mode
+            return executor.submit(self._run, *args, **kwargs).result()
+
+        # Non-Blocking call in async mode
+        return (
+            executor.submit(self._run, *args, **kwargs)
+            if executor
+            else self._run(*args, **kwargs)
+        )
+
+    @property
+    def num_async_workers(self) -> int:
+        """
+        :return: Number of threads of execution used for inference, returns 1
+            when multiple workers not specified
+        """
+        return self.num_async_workers
```

Comment:
This property is not really used anywhere, and can be removed if reviewers feel that way

Reply:
as discussed - we should remove any properties that are not pipeline constructor arguments